pkg/background/update_request_controller.go

```diff
@@ -257,34 +248,15 @@ func (c *Controller) updatePolicy(old, cur interface{}) {
 
 func (c *Controller) addUR(obj interface{}) {
 	ur := obj.(*kyvernov1beta1.UpdateRequest)
-	if ur.Status.Handler != "" {
-		return
-	}
 	c.enqueueUpdateRequest(ur)
 }
 
-func (c *Controller) updateUR(old, cur interface{}) {
-	oldUr := old.(*kyvernov1beta1.UpdateRequest)
+func (c *Controller) updateUR(_, cur interface{}) {
 	curUr := cur.(*kyvernov1beta1.UpdateRequest)
-	if oldUr.ResourceVersion == curUr.ResourceVersion {
-		// Periodic resync will send update events for all known Namespace.
-		// Two different versions of the same replica set will always have different RVs.
-		return
-	}
-	// only process the ones that are in "Pending"/"Completed" state
-	// if the UPDATE Request fails due to incorrect policy, it will be requeued during policy update
-	if curUr.Status.State != kyvernov1beta1.Pending {
-		return
-	}
```

Comment:
Can we add this back? I think there's no need to enqueue non-pending UR again on update.

Reply:
TBH, I would rather check this in the reconciler func.

Reply:
But this is true for UPDATE, not Add or DELETE.

Reply:
The intent was to reduce the frequency to update URs to avoid update conflicts.

Reply:
Ok, iadded it back.